installer/mod.ts

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env deno --allow-all
+
+const {
+  args,
+  env,
+  readDirSync,
+  mkdirSync,
+  writeFileSync,
+  exit,
+  stdin,
+  run,
+} = Deno;
+import * as path from '../fs/path.ts';
+import { parse } from './shebang.ts';
+
+enum Permission {
+  Unknown,
+  Read,
+  Write,
+  Net,
+  Env,
+  Run,
+  All,
+}
+
+function getPermissionFromFlag(flag: string): Permission {
+  switch (flag) {
+    case '--allow-read':
+      return Permission.Read;
+    case '--allow-write':
+      return Permission.Write;
+    case '--allow-net':
+      return Permission.Net;
+    case '--allow-env':
+      return Permission.Env;
+    case '--allow-run':
+      return Permission.Run;
+    case '--allow-all':
+      return Permission.All;
+    case '-A':
+      return Permission.All;
+  }
+  return Permission.Unknown;
+}
+
+function getFlagFromPermission(perm: Permission): string {
+  switch (perm) {
+    case Permission.Read:
+      return '--allow-read';
+    case Permission.Write:
+      return '--allow-write';
+    case Permission.Net:
+      return '--allow-net';
+    case Permission.Env:
+      return '--allow-env';
+    case Permission.Run:
+      return '--allow-run';
+    case Permission.All:
+      return '--allow-all';
+  }
+  return '';
+}
+
+async function readCharacter(): Promise<string> {
+  const decoder = new TextDecoder('utf-8');
+  const byteArray = new Uint8Array(1024);
+  await stdin.read(byteArray);
+  const line = decoder.decode(byteArray);
+  return line[0];
+}
+
+async function grantPermission(
+  perm: Permission,
+  moduleName: string = 'Deno'
+): Promise<boolean> {
+  let msg = `${moduleName} requests `;
+  switch (perm) {
+    case Permission.Read:
+      msg += 'read access to file system. ';
+      break;
+    case Permission.Write:
+      msg += 'write access to file system. ';
+      break;
+    case Permission.Net:
+      msg += 'network access. ';
+      break;
+    case Permission.Env:
+      msg += 'access to environment variable. ';
+      break;
+    case Permission.Run:
+      msg += 'access to run a subprocess. ';
+      break;
+    case Permission.All:
+      msg += 'all available access. ';
+      break;
+    default:
+      return false;
+  }
+  msg += 'Grant permanently? [yN]';
+  console.log(msg);
+
+  const input = await readCharacter();
+  if (input !== 'y' && input !== 'Y') {
+    return false;
+  }
+  return true;
+}
+
+function createDirIfNotExists(path: string) {
+  try {
+    readDirSync(path);
+  } catch (e) {
+    mkdirSync(path);
+  }
+}
+
+function getInstallerHome(): string {
+  const { DENO_DIR, HOME } = env();
+  if (!HOME && !DENO_DIR) {
+    throw new Error('$DENO_DIR and $HOME are not defined.');
+  }
+  if (DENO_DIR) {
```

Comment:
Using `DENO_DIR` instead of default `~/.deno/bin` will require to add `$DENO_DIR/bin/` to path as well to run scripts from shell - maybe we can skip this step for now?